src/test/java/uk/ac/ebi/eva/pipeline/io/VepProcessTest.java

```diff
@@ -0,0 +1,109 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.pipeline.io;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Matchers;
+import org.opencb.opencga.storage.mongodb.variant.DBObjectToVariantConverter;
+
+import uk.ac.ebi.eva.pipeline.model.VariantWrapper;
+import uk.ac.ebi.eva.pipeline.parameters.AnnotationParameters;
+import uk.ac.ebi.eva.test.data.VariantData;
+import uk.ac.ebi.eva.test.rules.PipelineTemporaryFolderRule;
+
+import java.io.File;
+
+import static uk.ac.ebi.eva.test.rules.TemporaryMongoRule.constructDbObject;
+import static uk.ac.ebi.eva.utils.FileUtils.getResource;
+
+public class VepProcessTest {
+
+    private AnnotationParameters annotationParameters;
+
+    @Rule
+    public PipelineTemporaryFolderRule temporaryFolder = new PipelineTemporaryFolderRule();
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    private static final int CHUNK_SIZE = 10;
+
+    private static final long VEP_TIMEOUT = 1;
+
+    private static final String HINT_CALL_OPEN = "hint: call open() before";
+
+    @Before
+    public void setUp() throws Exception {
+        annotationParameters = new AnnotationParameters();
+        annotationParameters.setFileId("fid");
+        annotationParameters.setStudyId("sid");
+        annotationParameters.setVepCacheVersion("1");
+        annotationParameters.setVepCachePath("cache");
+        annotationParameters.setVepPath(getResource("/mockvep_writeToFile.pl").getAbsolutePath());
+        annotationParameters.setVepCacheSpecies("hsapiens");
+        annotationParameters.setInputFasta("fasta");
+        annotationParameters.setVepNumForks("4");
+
+        File annotationFolder = temporaryFolder.newFolder();
+        annotationParameters.setOutputDirAnnotation(annotationFolder.getAbsolutePath());
+    }
+
+    @Test
+    public void testWorkflowWriteWithoutOpening() throws Exception {
+        DBObjectToVariantConverter converter = new DBObjectToVariantConverter();
+        VariantWrapper variantWrapper = new VariantWrapper(
+                converter.convertToDataModelType(constructDbObject(VariantData.getVariantWithAnnotation())));
+        annotationParameters.setVepPath(getResource("/mockvep_writeToFile_delayed.pl").getAbsolutePath());
+
+        VepProcess vepAnnotationFileWriter = new VepProcess(annotationParameters, CHUNK_SIZE, VEP_TIMEOUT);
+
+        exception.expect(IllegalStateException.class);
+        exception.expectMessage(Matchers.contains(HINT_CALL_OPEN));
```

Comment:
Checking message contents not externalized to configuration files is quite risky: changing a single character will break the test. If the test case set up is simplified (as in the previous comment) it should be enough to just expect an IllegalStateException for the whole method.